pkg/util/validation/limits.go

```diff
@@ -485,6 +486,7 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 
 	// Store-gateway.
 	f.IntVar(&l.StoreGatewayTenantShardSize, "store-gateway.tenant-shard-size", 0, "The tenant's shard size, used when store-gateway sharding is enabled. Value of 0 disables shuffle sharding for the tenant, that is all tenant blocks are sharded across all store-gateway replicas.")
+	f.IntVar(&l.StoreGatewayTenantShardSizePerZone, "store-gateway.tenant-shard-size-per-zone", 0, "The tenant's shard size per availability zone when zone-awareness is enabled, used when store-gateway sharding is enabled. The total shard size is computed as this value multiplied by the number of zones. This option is mutually exclusive with -store-gateway.tenant-shard-size.")
```

Comment:
I would improve the description saying that this takes precedence over `-store-gateway.tenant-shard-size`, instead of saying they're mutually exclusive (you can still set both, but this takes precedence).